DANS-KNAW/easy-bag-store
# EASY-1887 return 410 instead of a 409 or 500 when requesting inactive bag / prohibit requesting single items from inactive bag

Fixes EASY-1887

- [x]  merge with PR #81 first
- [x] Resolve merge conflicts

**It does not work for all paths leading to the enum function. So you can still get a list of the files in an inactive bag (but you can't get the content as text or zip).**
 

#### When applied it will...
* return a 410 instead of  500 when a hidden bag is requested trough bagsServletComponent
* return a 410 instead of 409 when a hidden bag is requested through bagStoreServletComponent
* prohibit to request individual items from an inactive bag through http
* prohibit to request individual items from an inactive bag through the stream command line option (unless the forceInactive param is supplied)


#### Where should the reviewer @DANS-KNAW/easy start?

#### How should this be manually tested?

steps

start deasy

ensure the latest version of easy-bag-store is built and deployed

ingest a bag to the bag store

connect to the deasy VM via SSH

set the bag to inactive using the CLI: easy-bag-store deactivate <bag-id>

run curl -v -H 'accept: application/zip' 'http://localhost:20110/bags/<bag-id>' to confirm the first problem

run curl -v -H 'accept: text/plain' 'http://localhost:20110/bags/<bag-id>/bagit.txt' to confirm the second problem

#### Related pull requests on github

repo                       | PR
-------------------------- | -----------------
easy-bag-store                   | [PR#77](https://github.com/DANS-KNAW/easy-bag-store/pull/77)
